controllers/node_kernel_reconciler.go

```diff
@@ -38,6 +42,11 @@ func (r *NodeKernelReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	kernelVersion := node.Status.NodeInfo.KernelVersion
+	osImage := osVersionRegexp.FindString(node.Status.NodeInfo.OSImage)
+	if osImage == "" {
+		return ctrl.Result{}, fmt.Errorf("could not get node %s osImage", node.Name)
+	}
+	syncronizedKernelOsDtkMapping.NewNodeInfo(kernelVersion, osImage)
```

Comment:
what about changing the predicate for this reconciler?

Reply:
Good catch. Forgot about it.

Reply:
Done.

Reply:
Done.